src/core/execute.c

```diff
@@ -2735,6 +2752,28 @@ static void invalid_env(const char *p, void *userdata) {
         log_unit_error(info->unit, "Ignoring invalid environment assignment '%s': %s", p, info->path);
 }
 
+int exec_context_named_iofds(Unit *unit, const ExecContext *c, const ExecParameters *p, int *named_iofds) {
+        unsigned int i, targets = 0;
+        targets += (c->std_input == EXEC_INPUT_NAMED_FD) ? 1 : 0;
+        targets += (c->std_output == EXEC_OUTPUT_NAMED_FD) ? 1 : 0;
+        targets += (c->std_error == EXEC_OUTPUT_NAMED_FD) ? 1 : 0;
+        for (i = 0; i < p->n_fds && targets > 0; i++) {
+                if (named_iofds[STDIN_FILENO] < 0 && c->std_input == EXEC_INPUT_NAMED_FD && streq(p->fd_names[i], "StandardInput")) {
```

Comment:
Note to self: this check that `fd < 0` means that second and subsequent descriptors with the same name will be ignored. This matches commit description.